config-provisioning/src/main/java/com/yahoo/config/provision/NetworkPorts.java

```diff
@@ -47,56 +44,14 @@ public String toString() {
         }
     }
 
-    private Map<Integer, Allocation> byPorts = new TreeMap<>();
-    private Map<String, Allocation> byKeys = new HashMap<>();
+    private final List<Allocation> allocations;
 
-    public Map<Integer, Allocation> byPortMap() {
-        return ImmutableMap.copyOf(byPorts);
-    }
-    public Map<String, Allocation> byKeyMap() {
-        return ImmutableMap.copyOf(byKeys);
-    }
-    public Collection<Allocation> allocations() {
-        return ImmutableMap.copyOf(byPorts).values();
+    public NetworkPorts(Collection<Allocation> allocations) {
+        this.allocations = new ArrayList<>(allocations.size());
```

Comment:
Can be simplified to `new ArrayList<>(allocations)`, or simply `List.copyOf` which also omits the need for `unmodifiableList` wrapping below.

Reply:
Or just ImmutableList also in the field declaration, which makes it explicit that the list won't change.